src/System.IO.Pipes/src/System/IO/Pipes/PipeStream.Unix.cs

```diff
@@ -292,9 +327,23 @@ internal static Interop.libc.OpenFlags TranslateFlags(PipeDirection direction, P
                 int fd = (int)handle.DangerousGetHandle();
                 Debug.Assert(fd >= 0);
 
-                long result;
-                while (Interop.CheckIo(result = sysCall(fd, arg1, arg2))) ;
-                return result;
+                while (true)
```

Comment:
I don't think so. The only way that'd happen is if every system call got interrupted each and every time.